datafusion/core/src/physical_plan/joins/symmetric_hash_join.rs

```diff
@@ -1119,6 +1159,144 @@ impl OneSideHashJoiner {
         Ok(())
     }
 
+    /// Gets build and probe indices which satisfy the on condition (including
+    /// the equality condition and the join filter) in the join.
+    #[allow(clippy::too_many_arguments)]
+    pub fn build_join_indices(
```

Comment:
The old implementation moved to symmetric hash join. The added complexity to support both options in a more generic way seems to add more complexity than just having the two versions around (and further tune them for the specific purpose / algorithm).